modules/lvms-using-raid-arrays.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="lvms-using-raid-arrays_{context}"]
+= Integrating software RAID arrays with {lvms}
+
+Logical Volume Manager (LVM) does not support creating a software RAID. However, you can create the RAID arrays using the `mdadm` utility, and integrate the RAID arrays with the {lvms}.
+
+.Prerequisites
+
+* You have created a software RAID during the {product-title} installation.
++
+[NOTE]
+====
+You can create the RAID arrays only during the installation of {product-title}. For information about creating RAID arrays in {product-title}, see link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/installing/installation-configuration#installation-special-config-raid_installing-customizing[Configuring a RAID-enabled data volume]
+====
+
+* You have installed {lvms}.
+
+.Procedure
+
+. Open the `LVMCluster` custom resource (CR) YAML file.
++
+[NOTE]
+====
+If you have created the `LVMCluster`, then you can edit the existing `LVMCluster` CR. Else, you can perform this step while creating a new `LVMCluster`.
+====
+
+. Add the path to the RAID array device in the `deviceSelector` section in `LVMCluster` CR YAML.
++
+[source,yaml]
+----
+apiVersion: lvm.topolvm.io/v1alpha1
+kind: LVMCluster
+metadata:
+  name: my-lvmcluster
+spec:
+  storage:
+    deviceClasses:
+    - name: vg1
+      default: true
+      deviceSelector: <1>
+        paths:
+        - /dev/disk/by-path/pci-0000:87:00.0-nvme-1
+        - /dev/disk/by-path/pci-0000:88:00.0-nvme-1
+        - /dev/md/md-var <2>
```

Comment:
we should include an example for an autogenerated md device as well:
`/dev/md0 <2>`

Reply:
I doubt if we should recommend adding the RAID device at the time of creating the CR. Because editing the CR, though it can be a complex or lengthy process, it is still possible. If the user wants to intentionally skip adding the RAID devices, and decides to add them only when required in the future, in this case editing the CR is the only choice. 

We also have procs like "Scaling up storage of clusters" which are all about editing the existing CRs. In this case it is best to just mention both the possibilities (adding by editing the CR, and adding at the time of creating the CR), and users can choose what works best for them.

What do you think of this approach?

Reply:
Could you provide a bit more info on the autogenerated RAID array device?
Just to describe the purpose of these devices.

Reply:
> I doubt if we should recommend adding the RAID device at the time of creating the CR. Because editing the CR, though it can be a complex or lengthy process, it is still possible. 

That is true, but configuring a RAID array is something possible only at install time when configuring the nodes. That means that setting up LVMCluster with the mdadm device paths is also a install-time operation and the docs should cover the 90% case.

> If the user wants to intentionally skip adding the RAID devices, and decides to add them only when required in the future, in this case editing the CR is the only choice.

This is correct, but again, speaking from our customers, an edge case.

> We also have procs like "Scaling up storage of clusters" which are all about editing the existing CRs. In this case it is best to just mention both the possibilities (adding by editing the CR, and adding at the time of creating the CR), and users can choose what works best for them.

I agree, because this process is a typical day 2 operation for clusters, unlike configuring a storage device initially on node start.

> What do you think of this approach?

Fine with both as long as they work correctly.

> Could you provide a bit more info on the autogenerated RAID array device?
Just to describe the purpose of these devices.

Sure. If one creates an mdadm device via `mdadm --create /dev/md/<Label>`, then it will show as you also described in the example. However, if you leave this away and just use a device path like `mdadm --create /dev/md0 ...` then the device will not be unter /dev/md/label but directly as /dev/md0 instead. Just a nit, not to worry too much about it

Reply:
@jakobmoellerdev 
I have added a recommendation as suggested. I've also included a line in the "Creating LVMCluster" proc - in "deviceSelector" field description to indicate that users can add RAID array paths at the time of creating the cluster.

I have added the example for "/dev/md0" but did not use the term "autogenerated" to describe the RAID array as it is a new term and it might need additional description which is not in the scope of this module.